src/crypto/chacha20_vec.ipp

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) 2025-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <crypto/chacha20_vec.h>
+
+#include <bit>
+#include <cassert>
+#include <cstring>
+#include <limits>
+
+#if defined(ENABLE_CHACHA20_VEC)
+
+#if defined(CHACHA20_VEC_DISABLE_STATES_16) && \
+    defined(CHACHA20_VEC_DISABLE_STATES_8) && \
+    defined(CHACHA20_VEC_DISABLE_STATES_6) && \
+    defined(CHACHA20_VEC_DISABLE_STATES_4) && \
+    defined(CHACHA20_VEC_DISABLE_STATES_2)
+#define CHACHA20_VEC_ALL_MULTI_STATES_DISABLED
+#endif
+
+
+#if !defined(CHACHA20_VEC_ALL_MULTI_STATES_DISABLED)
+
+#if defined(__has_attribute)
+#  if __has_attribute(always_inline)
+#    define ALWAYS_INLINE __attribute__ ((always_inline)) inline
+#  endif
+#endif
+
+#if !defined(ALWAYS_INLINE)
+#  define ALWAYS_INLINE inline
+#endif
+
+
+namespace {
+
+using vec256 = uint32_t __attribute__((__vector_size__(32)));
```

Comment:
`static_assert(sizeof(vec256) == 32)` ? (Or `sizeof(vec256) == 32 || ALL_MULTI_STATES_DISABLED`)

Reply:
Sorry, I meant just so you get an error if you're using a compiler ignores the `__attribute__` entirely but still passes the `#if` in _base.cpp.